### PR TITLE
Constitution 2020 update

### PR DESCRIPTION
**Pull request by haydenstec**
Summary of Proposed Revisions:

- Revised Supervisors and removed initiatives
- Clarified election day process by traditionally used method (Start election of positions in order of priority, give each candidate time to speak and answer questions, vote, then move to next position and repeat).
- Added new requirements for period leading up to election to improve transparency and encourage a more lively election
- Restricted "Coordinator" positions to only Freshmen and Sophomores
- Removed Outreach Coordinator position
- Normalized references to the club in Constitution to "Pitt CSC"
- Clarified current officer positions in light of the last two years of club progress (EX: Event Coordinator frequently makes or contracts someone to make promotional material).
- Added new language to describe future possible developments of current officer positions given their recent development (EX: Business Manager has been the chief record keeper in a sense these last two years, therefore it may come to pass that they might survey the club in the future to get input on activities, an overall review of how a semester has gone, etc.).
- Added provision to allow special elections in case of vacancy.
- Added a Presidential Farewell Address ("There, the President will address the general body and commence elections").
- Added former officers as student advisors

### CODE REVIEW

File: Constitution.sublime-workspace [hunk 1]
```diff
@@ -0,0 +1,651 @@
+{
```
Comment (varughese) [inline Constitution.sublime-workspace]: Haha, should probably delete this file / add it to your .gitignore

**Comment by rohit-ganguly** [comment]
hi hayden

File: Constitution.md [hunk 2]
```diff
@@ -85,113 +86,126 @@ Officers will begin their terms in the Spring semester and will continue until t
...
+
+The Media Coordinator is responsible for managing all Pitt CSC social media accounts and email. He or she will design posts for Pitt CSC social media, and will attend all Pitt CSC events, taking photos, videos, and any testimony he or she deems helpful for creating effective social media posts.
+
+### Project Coordinator
 
-### Outreach Coordinator
+The Project Coordinator is responsible for organizing and managing club projects, facilitating personal and small group projects among members, and bringing Open Source or other external projects to Pitt CSC's attention. He or she will work with the Event Coordinator to plan relevant activities, and with the President and/or Vice President when interacting with open source or other external groups/projects. He or she will also directly survey and work with the membership to aid in planning, development, and deployment of club projects.
 
-The Outreach Coordinator is repsonsible for representing Pitt CSC to outside organizations. He or she will coordinate with organizations on behalf of Pitt CSC with regards to planning events, seeking sponsorships and community partnerships, etc.
+### Lord Protector
```
Comment (azharichenko) [inline Constitution.md]: 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔

Comment (RitwikGupta) [reply Constitution.md]: Why not call this "Officer Emeritus" or something? Also, why is this necessary? It's a CS club, I don't see what an honorary position would really accomplish.

File: Constitution.md [hunk 3]
```diff
@@ -85,113 +86,126 @@ Officers will begin their terms in the Spring semester and will continue until t
...
 
+The Lord Protector is an honorary position for a former Pitt CSC officer, appointed by a vote of all former officers who maintain student status.
 
 ## XIV) Election of Officers
 
 Elections will be held in each Spring semester to determine the officers for the proceeding Fall and Spring terms.
+Only freshmen and sophomores are permitted to run for Event Coordinator, Project Coordinator, and Media Coordinator. Should no freshmen or sophomores declare the intent to run for any of these positions by the deadline to declare intent to run, then any member will be permitted to run.
+The President will preside over elections, announcing them to the general body at least one month prior to the election date via the email list and Pitt CSC Slack. Any candidate must inform the President of their intent to run at least one week before the scheduled date of the election. Prior to the election day, the President confirm that all candidates are Pitt CSC members according to the requirements set out in this Constitution. Following confirmation, all valid candidates will be announced by the President to the general body before the election date.
```
Comment (RitwikGupta) [inline Constitution.md]: To be more flexible to future communication methods, I would generalize the "email list and Pitt CSC Slack" language.

File: Constitution.md [hunk 4]
```diff
@@ -85,113 +86,126 @@ Officers will begin their terms in the Spring semester and will continue until t
...
+During a general body meeting, any member may call for a motion for the defense of Pitt CSC. This motion must contain a reason for entering a period of defense of Pitt CSC, and a time frame no longer than one academic year for the period of defense of Pitt CSC. If seconded, an immediate, public, three fourths majority vote is held on the matter. Should it pass, the Lord Protector is installed as the sole member of the executive board for the time frame, and is personally exempt from the restriction on holding multiple officer positions. No constitutional amendments are allowed to be passed during a period of defense of Pitt CSC. During the period of defense of Pitt CSC, the Lord Protector is tasked with effectively reorganizing and managing Pitt CSC in line with its stated purpose and mission, and with resolving whatever conflict or issue was posed in the original call for the period. After the time frame ends, or with the approval of the Lord Protector, normal club management will resume with Officers being reinstated as executive board members and the Lord Protector being removed.
```
Comment (RitwikGupta) [inline Constitution.md]: What is going on here, why is the Pitt CSC writing Roman Empire-like dictatorial powers into the constitution? What kind of crisis could possibly arise where a dictatorial former officer needs to take over?

**Pull request merged**